Update to the definition of span tag and attributes (#26231)
DataDog/documentation: content/en/tracing/trace_collection/tracing_naming_convention/_index.md

Span tags and span attributes are similar but distinct concepts:

> - [Span tags](#span-tags) are the enrichments of context related to the span. For instance, host or container tags describing the infrastructure the service is running on.

Reviewer: ```suggestion
- [Span tags](#span-tags) provides context related to the span. For instance, host or container tags on the infrastructure the service is running on.
```

- [Span attributes](#span-attributes) are the content of the span, collected with automatic or manual instrumentation in the application.

### Span tags

> Span tags are the enrichments of context related to the span. For instance, host or container tags describing the infrastructure the service is running on. More examples include:

Reviewer: ```suggestion
Span tags provide context related to the span. For instance, host or container tags on the infrastructure the service is running on. More examples include:
```
